README.md

```diff
@@ -4,111 +4,209 @@ An API for validating survey schemas.
+
+Only the following URLs and domains are accepted:
+- "https://gist.githubusercontent.com/"
+- "https://raw.githubusercontent.com/"
+- "onsdigital.uk"
+- "localhost"
+
+Also when using a URL from GitHub you can only validate schemas from the ONSdigital organisation and only against repos with the owner "ONSdigital".
+
+### Running against eQ Runner
+
+Also once you have validator running it can be used to run against eQ runner (`https://github.com/ONSdigital/eq-questionnaire-runner`). If you have runner spun up you can run the `make validate-test-schemas` script from within the root of runner which will run validator on the test runner schemas.
+
+### Running the Ajv (server) version of validator
+
+Then run the Ajv server:
+``` shell
+make start-ajv
 ```
-http://localhost:5001/validate?url=...
+
+This defaults to running on port `5002` set `AJV_VALIDATOR_PORT` in your .env file if you need to change this.
+
+To stop the Ajv server:
+``` shell
+make stop-ajv
 ```
 
+Running this returns either an empty json response when the questionnaire is valid, or a response containing an "errors" key. The errors are ordered by their path length and with first error message being the deepest path into the schema and should represent the best match for the questionnaire which has been posted.
 
-## Testing
+### Testing and running against local schemas
 
-By default, all schemas in the `tests/schemas/valid` directory will be evaluated as part of the unit tests.
-Any errors in these schemas will cause a failure.
+By default, all schemas in the `tests/schemas/valid` and `tests/schemas/invalid` directories will be evaluated as part of the unit tests. Any errors in these schemas will cause a failure.
 
 To run the app's unit tests:
-
-```
+``` shell
 make test-unit
 ```
 
-To run the app's unit tests and ajv validator tests:
+Make sure you don't already have Ajv running on localhost:5002 by running `lsof -i tcp:5002` if you do make a note of the PID and then run `kill -9 PID`, replacing "PID" with the process id from the previous command.
 
+Run the Ajv validator tests:
+``` shell
+make test-ajv
 ```
+
+To run the app's unit tests and Ajv validator tests:
+``` shell
 make test
 ```
 
-To test the apps functionality:
-```
+#### Test the local validator app against runner schemas
+
+Spin validator up with:
+``` shell
 make run
 ```
 
-Then, in another terminal window/tab, navigate to a checked out copy of eq-survey-runner:
+Then, in another terminal, navigate to a checked out copy of `https://github.com/ONSdigital/eq-questionnaire-runner` and run:
+``` shell
+make validate-test-schemas
 ```
-make test
+This will run the validator against all runner test schemas.
+
+Or you can run it against a specific runner schema, to do this:
+- set the runner vars:
+    - `SCHEMA_PATH` to the path of the schema file (defaults to `./schemas/test/en/`)
+    - `SCHEMA` to the schema file name without the `.json`
+- then run:
+``` shell
+make validate-test-schema
```

Comment:
Very minor, and just an alternative way of doing it, but could set these vars inline with something like `make validate-test-schema SCHEMA=test_checkbox`